packages/fxa-auth-server/scripts/recorded-future/check-and-reset.ts

```diff
@@ -0,0 +1,115 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A script to check Recorded Future's Identity API for leaked credential, and
+ * force an account reset on accounts with valid leaked credentials.
+ *
+ * This script relies on the same set of enivronment variables as the FxA auth
+ * server.
+ */
+
+/**
+ * The general steps are:
+ *   1. Query Recorded Future's Identity credential search API endpoint to
+ *   fetch new leaked credential since the previous check.
+ *   2. For logins that are email addresses with an account in FxA, query
+ *   credential lookup API endpoint to get the leaked credentials.
+ *   3. For accounts with valid leaked plaintext passwords and no TOTP 2FA,
+ *   reset the account.
+ *   4. Send an email to the affected account holder to reset their password.
+ */
+
+import { promisify } from 'util';
+
+import { Command } from 'commander';
+import { StatsD } from 'hot-shots';
+import createClient from 'openapi-fetch';
+
+import { parseBooleanArg } from '../lib/args';
+import { emitStatsdMetrics } from '../lib/metrics';
+import appConfig from '../../config';
+
+import { paths } from './identity-schema';
+
+const config = appConfig.getProperties();
+const searchDomain = 'accounts.firefox.com';
+const statsd = new StatsD({ ...config.statsd });
+const client = createClient<paths>({
+  baseUrl: 'https://api.recordedfuture.com',
+  headers: { 'X-RFToken': config.recordedFuture.identityApiKey },
+});
+
+const checkAndReset = async () => {
+  const program = new Command();
+  program
+    .description(
+      `Query Recorded Future's Identity API for leaked account credentials.  Accounts
+with valid leaked credentials and without TOTP 2FA will be reset.  An email
+will be sent to the account holder.`
+    )
+    .option(
+      '--first-downloaded-date <date>',
+      'The date after when the account credential was first downloaded by Recorded Future.  If not given, the date in UTC from 24 hours ago will be used.',
+      Date.parse
+    )
+    .option(
+      '--dry-run [true|false]',
+      'Print out the argument and configuration values that will be used in the execution of the script. Defaults to true.',
+      parseBooleanArg,
+      true
+    );
+
+  program.parse(process.argv);
+
+  const firstDownloadedDatetime =
+    program.firstDownloadedDate && !Number.isNaN(program.firstDownloadedDate)
+      ? new Date(program.firstDownloadedDate)
+      : new Date(Date.now() - 24 * 60 * 60 * 1000);
+  const firstDownloadedDateIsoString = firstDownloadedDatetime
+    .toISOString()
+    .split('T')[0];
+
+  if (program.dryRun) {
+    console.log(`
+Dry run mode is on.  It is the default; use '--dry-run false' when you are ready.
+
+Domain: ${searchDomain}
+Filter: first_downloaded_date_gte=${firstDownloadedDateIsoString}
+Limit: 500
+`);
+    return 0;
+  }
+
+  const res = await emitStatsdMetrics(
```

Comment:
I like the pattern in `enqueue-inactive-account-deletions.ts` where the function is extracted, e.g.,

```
const credentialSearch = await emitStatsdMetrics(
  _credentialSearch,
  'recorded-future.identity.credentials-search',
  statsd
)();
```